src/txn/txn_rollback_to_stable.c

```diff
@@ -156,6 +158,16 @@ __txn_abort_row_replace_with_hs_value(
 
     WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
 
+    /* Get the full update value from the data store. */
+    unpack = &_unpack;
+    __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
+    WT_CLEAR(ds_value);
+    if (!__wt_row_leaf_value(page, rip, &ds_value))
+        WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &ds_value));
+
+    WT_CLEAR(full_value);
+    WT_ERR(__wt_buf_set(session, &full_value, ds_value.data, ds_value.size));
```

Comment:
I'm not sure about the pattern of calls in rollback-to-stable, but would we be better served by creating a normal cursor on the data store, rather than rolling our own code here?

@kommiharibabu?

Reply:
@keithbostic The main reason behind the not using the cursor API is as we don't have a way to provide our own transaction id and timestamp that is retrieved from the history store or data store. All the rollback of the on-disk abort scenarios, where we place an update in the update list that is lesser timestamp than the on-disk, so we should handle some corner cases like, we shouldn't let the on-disk value to be restored into the update chain and etc.